plutus-playground-lib/src/Schema.hs

```diff
@@ -0,0 +1,320 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE DefaultSignatures          #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+
+-- | For the Playground, we need to be able to represent function
+-- signatures, and function calls, as JSON objects.
+--
+-- 'ToSchema' turns a function signature into a 'FormSchema' - a
+-- description that can serialised as JSON and analysed in PureScript
+-- land. In essence it's a simplified, specialised version of a
+-- 'Generic' representation.
+--
+-- The frontend then takes a 'FormSchema', generates a UI form, and
+-- allows the user create a concrete value that follows that
+-- schema. This is the 'FormArgument'.
+--
+-- It's useful for the backend to make this instantiation too (because
+-- we want to give the user example filled-in forms), so we provide
+-- ToArgument.
```

Comment:
:+1: Could we also get Haddock for the major definitions (at least the classes)?

Reply:
I realise this module is mostly moved, so this isn't new, but still.

Reply:
Yes, good call. Done. I expanded the docs for `IOTS` too while I'm at it.